internal/commands/app/create_inputs.go

```diff
@@ -131,6 +135,65 @@ func (i *createInputs) resolveLocalPath(ui terminal.UI, wd string) (string, erro
 	return fullPath, nil
 }
 
+func (i *createInputs) resolveTemplateID(ui terminal.UI, client realm.Client, templateID string) (string, error) {
+	if templateID == "" && ui.AutoConfirm() {
+		return "", nil
+	}
+
+	templates, err := client.Templates()
+	if err != nil {
+		return "", err
+	}
+
+	// do not disrupt application creation flow if templates are not
+	// available and user is not specifying a template
+	if templateID == "" && len(templates) == 0 {
+		return "", nil
+	}
+
+	if len(templates) == 0 {
+		return "", fmt.Errorf("unable to find template '%s'", templateID)
+	}
+
+	var resolvedTemplateID string
+	if templateID == "" {
+		options := make([]string, 0, len(templates)+1)
+		templateIDs := make([]string, 0, len(templates)+1)
+		options = append(options, "[No Template]: Do Not Use A Template")
```

Comment:
yeah, but "every time" here is really only when a user is creating a new app, which seems completely fine here. otherwise, if a user is using `-y`, we skip all of this, unless said `-y` user has specified a `--template` param.

the other option is to **not** automatically/interactively prompt users to select a template, but I feel like that makes the template apps feature less discoverable for CLI users. @andrewzlchen might have more context on that requirement though

Reply:
Hmm i think we should prompt users on the template options on every `realm-cli app create` invokation

Reason for that is that the objective of this project to help users who are starting out with realm to have a gentle intro experience. 

Hiding the templates by default kinda goes against that project objective.

This was also kinda why i wanted to use the `--use-template` flag because we can skip fetching the templates in the event the user says no which may be easier on the server

The bottom line is that we should definitely prompt for whether the user wants to create their app with a template - whether it's with the --use-template flag or the just the --template

Reply:
Not sure if I'm misinterpreting your intent, but my understanding of the CLI was that it would prompt in the event that the user does not pass in a required flag. If this is true then we would need to prompt if the user does not supply the template parameter out of necessity.

Also, I don't really want to introduce new concepts here by introducing a global flag tbh

Maybe the happy middle ground here is to have the `--use-template` (or a `--no-template` since --use-template is only needed if we didn't want to use a template)